sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -2911,4 +2911,28 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
   override def visitRefreshTable(ctx: RefreshTableContext): LogicalPlan = withOrigin(ctx) {
     RefreshTableStatement(visitMultipartIdentifier(ctx.multipartIdentifier()))
   }
+
+  /**
+   * A command for users to list the column names for a table.
+   * This function creates a [[ShowColumnsStatement]] logical plan.
+   *
+   * The syntax of using this command in SQL is:
+   * {{{
+   *   SHOW COLUMNS (FROM | IN) tableName=multipartIdentifier
+   *        ((FROM | IN) namespace=multipartIdentifier)?
+   * }}}
+   */
+  override def visitShowColumns(ctx: ShowColumnsContext): LogicalPlan = withOrigin(ctx) {
+    import org.apache.spark.sql.connector.catalog.CatalogV2Implicits._
+
+    val table = visitMultipartIdentifier(ctx.table)
+    val namespace = Option(ctx.namespace).map(visitMultipartIdentifier)
+    if (namespace.isDefined && namespace.get.length > 1 && table.length > 1) {
```

Comment:
I think the logic should be: if the namespace is specified, then table name length must be 1.

Reply:
OK I get it. Previously we support `SHOW COLUMNS FROM db1.t1 IN db1`, and at runtime we check if the specified database name is the same as the db name in table identifier.

Seems like we should move this check into the analyzer, i.e. in `ResolveSessionCatalog`

Reply:
Since we don't have v2 SHOW COLUMNS yet, so in `ResolveSessionCatalog`, we should simply check that
1. the table name has <= 2 parts
2. the namespace name should have only one part if specified